reference/api/overview.md

```diff
@@ -1,45 +1,115 @@
+- [JSON](https://bump.sh/doc/meilisearch.json)
+
+## Document conventions
+
+The API documentation uses the following conventions:
+
+- Curly braces (`{}`) in API routes represent path parameters, e.g., GET `/indexes/{index_uid}`
+- Required fields are marked by an asterisk (`*`)
+- Placeholder text is in uppercase characters with underscore delimiters, e.g., `MASTER_KEY`
+
+## Authorization
+
+By [providing Meilisearch with a master key at launch](/learn/security/master_api_keys.md#protecting-a-meilisearch-instance), you protect your instance from unauthorized requests. From then on, you must include the `Authorization` header along with a valid API key to access protected routes (all routes except `/health`).
+
+<CodeSamples id="authorization_header_1" />
+
+The [`/keys`](/reference/api/keys.md) route can only be accessed using the master key. For security reasons, we recommend using regular API keys for all other routes.
+
+::: note
+ v0.24 and below use the `X-MEILI-API-KEY: apiKey` authorization header:
+<CodeSamples id="updating_guide_check_version_old_authorization_header" />
 :::
 
+[To learn more about keys and security, refer to our dedicated guide.](/learn/security/master_api_keys.md)
+
+## Pagination
+
+Meilisearch paginates all GET routes that return multiple resources, e.g., GET `/indexes`, GET `/documents`, GET `/keys`, etc. This allows you to work with manageable chunks of data. All these routes return 20 results per page, but you can configure it using the `limit` query parameter. You can move between pages using `offset`.
+
+All paginated responses contain the following fields:
+
+| Name         | Type    | Description                  |
+| :----------- | :------ | :--------------------------- |
+| **`offset`** | Integer | Number of resources skipped  |
+| **`limit`**  | Integer | Number of resources returned |
+| **`total`**  | Integer | Total number of resources    |
+
+### `/tasks` endpoint
+
+Since the `/tasks` endpoint uses a different type of pagination, the response contains different fields. You can read more about it in the [tasks API reference](/reference/api/tasks.md#get-tasks).
+
+## Parameters
+
+Parameters are options passed to the endpoint to influence the response. They are added at the end of the string for GET requests. For POST requests, they are added to the body.
+
+### Path parameters
+
+These are parameters you pass to the API in the endpoint's path. They are used to identify a resource uniquely. You can have multiple path parameters, e.g., `/indexes/{index_uid}/documents/{document_id}`.
+
+We omit this section if an endpoint does not take any path parameters.
+
+### Query parameters
+
+These optional parameters are a sequence of key-value pairs and appear after the question mark (`?`) in the endpoint. You can list multiple query parameters by separating them with an ampersand (`&`). The order of query parameters does not matter.
+
+We omit this section if an endpoint does not take any query parameters.
```

Comment:
I think it's good to specify where exactly we omit the query parameters section.

```suggestion
If an endpoint does not take any query parameters, this section is not present in that endpoint's documentation.
```